src/Planner/CollectTableExpressionData.cpp

```diff
@@ -192,13 +193,41 @@ void checkStorageSupportPrewhere(const QueryTreeNodePtr & table_expression)
     }
 }
 
+// checks that there are only user configured columns in requested set
```

Comment:
yes, at this point I tried to prohibit streaming queries with virtual columns, because by default for all tables such columns will be set in the TableSource, but in streaming queries, the blocks that the SubscriptionSource receives are user input blocks.